lib/Analysis/SelectVariableNames/SelectVariableNames.h

```diff
@@ -4,15 +4,54 @@
 #include <cassert>
 #include <string>
 
-#include "llvm/include/llvm/ADT/DenseMap.h"          // from @llvm-project
-#include "mlir/include/mlir/IR/BuiltinAttributes.h"  // from @llvm-project
-#include "mlir/include/mlir/IR/Operation.h"          // from @llvm-project
-#include "mlir/include/mlir/IR/Value.h"              // from @llvm-project
-#include "mlir/include/mlir/Support/LLVM.h"          // from @llvm-project
+#include "llvm/include/llvm/ADT/DenseMap.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/SparseAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlowFramework.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinAttributes.h"        // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"                // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                    // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"                // from @llvm-project
 
 namespace mlir {
 namespace heir {
 
+class VariableName {
+ public:
+  VariableName() = default;
+  VariableName(const std::string &name, int integer)
+      : initialized(true), name(name), integer(integer) {}
+  ~VariableName() = default;
+
+  bool operator==(const VariableName &other) const {
+    return initialized == other.initialized && name == other.name &&
+           integer == other.integer;
+  }
+
+  // For Lattice in DataFlowFramework
+  static VariableName join(const VariableName &lhs, const VariableName &rhs) {
+    if (!lhs.isInitialized() || lhs.getName().empty()) return rhs;
+    if (!rhs.isInitialized() || rhs.getName().empty()) return lhs;
+
+    if (lhs.getName() == rhs.getName()) return lhs;
+
+    // If the names are different, we can just return the first one
```

Comment:
I'm guessing this line never actually executes, if the transfer function below is a no-op... Maybe it would be worth adding an assertion here? Otherwise, I would question the use of this analysis that trivially ignores half the information.